sdk-base/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationPlugin.kt

```diff
@@ -3,12 +3,11 @@ package com.mapbox.maps.plugin.annotation
 import android.view.View
 import com.mapbox.maps.plugin.MapPlugin
 import com.mapbox.maps.plugin.MapSizePlugin
-import com.mapbox.maps.plugin.MapStyleObserverPlugin
 
 /**
  * Plugin interface for the annotation.
  */
-interface AnnotationPlugin : MapPlugin, MapSizePlugin, MapStyleObserverPlugin {
+interface AnnotationPlugin : MapPlugin, MapSizePlugin {
```

Comment:
If I remember correctly although we do not use `MapStyleObserverPlugin` - we still decided to leave it here but do not do any actions for now. Is this correct, @pengdev?

Reply:
yes, until we find a valid use case for it 😄

Reply:
But AnnotationPlugin doesn't need MapStyleObserverPlugin anymore. We can add it back if needed in the future.

Reply:
It was discussed by me and @pengdev. If we remove it from location component plugin and annotations plugin - it means that `MapStyleObserverPlugin` is redundant and could be fully removed from codebase. However it's a breaking change then and potentially this observer interface may be useful in future. That's why it was decided to let location component and annotation plugins implement it but atm do not perform any particular actions on style change callback.

Reply:
I see, will roll back this change.